timeplus/client.go

```diff
@@ -292,20 +283,48 @@ func (s *TimeplusClient) queryStreamV2(sql string, batchCount int, batchBufferTi
 	config := utils.NewDefaultHTTPClientConfig()
 	res, err := utils.SSEHttpRequestWithAPIKey(http.MethodPost, createQueryUrl, query, config, s.apikey)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create query : %w", err)
+		return nil, nil, nil, fmt.Errorf("failed to create query : %w", err)
 	}
 
 	reader := bufio.NewReader(res.Body)
 	ch := make(chan rxgo.Item)
-	canceled := false
-	go func() {
-		defer res.Body.Close()
+	var header QueryInfo
 
-		for {
-			if canceled {
+	// Read the header here
+	for {
+		line, err := readCompleteLine(reader)
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("failed to retrieve query header: %w", err)
+		}
+		if len(line) == 0 {
+			continue
+		}
+		colonIndex := strings.Index(line, ":")
+		eventField := strings.TrimSpace(line[0:colonIndex])
+		eventData := strings.TrimSpace(line[colonIndex+1:])
+
+		if eventField == "event" {
```

Comment:
we should check the other side: `if eventField != "event" return something is wrong`